libs/vectorise/include/vectorise/fixed_point/fixed_point.hpp

```diff
@@ -447,14 +447,7 @@ class FixedPoint : public BaseFixedpointType
   template <typename T>
   static constexpr bool CheckNoOverflow(T n)
   {
-    if (Type(n) <= MAX && Type(n) >= MIN)
-    {
-      return true;
-    }
-    else
-    {
-      return false;
-    }
+    return Type(n) <= MAX && Type(n) >= MIN;
```

Comment:
This is just suggestion since we are changing it here - following  should improve readability and should performance too (I think compiler optimization won't be able to detect, in general, that it does not need to create 2 instances of Type).

```suggestion
    auto const Type x{n};
    return MIN <= x && x <= MAX;
```